tests/test_object.h

```diff
@@ -87,6 +87,97 @@ TEST_CASE("[Object] Metadata") {
 			meta_list2.size() == 0,
 			"The metadata list should contain 0 items after removing all metadata items.");
 }
+
+TEST_CASE("[Object] IAPI methods") {
+	Object object;
+    bool *rvalid = memnew(bool(false));
+
+    object.set(StringName("script"), "script_source", rvalid);
+	CHECK_MESSAGE(
+			*rvalid == true,
+			"set() should update the passed bool to the expected value.");
+
+    *rvalid = false;
+	CHECK_MESSAGE(
+			object.get(StringName("script"), rvalid) == "script_source",
+			"get() should return the expected value.");
+	CHECK_MESSAGE(
+			*rvalid == true,
+			"get() should update the passed bool to the expected value.");
+    
+    Vector<StringName> p_names;
+    object.set_indexed(p_names, "value", rvalid);
+    CHECK_MESSAGE(
+		*rvalid == false,
+		"set_indexed() should update the passed bool to the expected value for empty Vector<StringName>.");
+
+    *rvalid = true;
```

Comment:
Sanity check: should this be reset to `false` (at least for consistency)?

```suggestion
    *rvalid = false;
```